### ISSUE TITLE
Android Permissions: Add the following line to your app's AndroidManifest.xml

### ISSUE THREAD

**Issue by dreallday** [react-native-community/react-native-maps#2213]
### Is this a bug report?

Maybe

### Have you read the [Installation Instructions](https://github.com/react-community/react-native-maps/blob/master/docs/installation.md)?

Yes

### Steps to Reproduce

Start the app.

### Expected Behavior

No crash.

### Actual Behavior

App Crash due to permission error.

`
04-23 11:24:09.959 9115-9209/---package name--- E/unknown:ReactNative: Exception in native call
    java.lang.SecurityException: Looks like the app doesn't have the permission to access location.
    Add the following line to your app's AndroidManifest.xml:
    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
        at com.facebook.react.modules.location.LocationModule.throwLocationPermissionMissing(LocationModule.java:234)
        at com.facebook.react.modules.location.LocationModule.startObserving(LocationModule.java:173)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
        at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:160)
        at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
        at android.os.Handler.handleCallback(Handler.java:789)
        at android.os.Handler.dispatchMessage(Handler.java:98)
        at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:29)
        at android.os.Looper.loop(Looper.java:164)
        at com.facebook.react.bridge.queue.MessageQueueThreadImpl$3.run(MessageQueueThreadImpl.java:192)
        at java.lang.Thread.run(Thread.java:764)
     Caused by: java.lang.SecurityException: "gps" location provider requires ACCESS_FINE_LOCATION permission.
        at android.os.Parcel.readException(Parcel.java:1942)
        at android.os.Parcel.readException(Parcel.java:1888)
        at android.location.ILocationManager$Stub$Proxy.requestLocationUpdates(ILocationManager.java:732)
        at android.location.LocationManager.requestLocationUpdates(LocationManager.java:914)
        at android.location.LocationManager.requestLocationUpdates(LocationManager.java:472)
        at com.facebook.react.modules.location.LocationModule.startObserving(LocationModule.java:165)
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372) 
        at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:160) 
        at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method) 
        at android.os.Handler.handleCallback(Handler.java:789) 
        at android.os.Handler.dispatchMessage(Handler.java:98) 
        at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:29) 
        at android.os.Looper.loop(Looper.java:164) 
        at com.facebook.react.bridge.queue.MessageQueueThreadImpl$3.run(MessageQueueThreadImpl.java:192) 
        at java.lang.Thread.run(Thread.java:764)
`
![screenshot_20180423-112412](https://user-images.githubusercontent.com/3266324/39136774-fe10cd9a-46e9-11e8-861c-0c04e02fd3cf.jpg)

### Android Manifest File

Contains the right permissions.

![screen shot 2018-04-23 at 11 28 30](https://user-images.githubusercontent.com/3266324/39136797-09befd74-46ea-11e8-91d0-ff0dd77c0abe.jpg)

### Environment
    "react": "^16.3.2",
    "react-native": "0.55.2",
    "react-native-maps": "^0.21.0",

###  TL;DR

Android crashes because missing permissions, but permissions aren't missing in the app's manifest file.

**Comment by rborn** [comment]
@dreallday could you please make a demo repo so we can test it?  Thank you.

**Comment by erick2014** [comment]
@dreallday did you try generating another key from google console? maybe that helps

**Comment by dreallday** [comment]
@erick2014 yes, didn't help.

@rborn that will take a minute

**Comment by MarkOSullivan94** [comment]
@dreallday where is your AndroidManfiest located (give the exact path starting from where your project is location) e.g. ../<project-name>/.../.../../

**Comment by dreallday** [comment]
I figured this out, it's due to react-native-maps having a higher deployment target than 23.
```
ext {
    compileSdkVersion   = 26
    targetSdkVersion    = 26
    buildToolsVersion   = "26.0.2"
    supportLibVersion   = "26.1.0"
    googlePlayServicesVersion = "11.8.0"
    androidMapsUtilsVersion = "0.5+"
}
```
> On devices before SDK version 23, the permissions are automatically granted if they appear in the manifest, so check and request should always be true.

Had to implement this: https://facebook.github.io/react-native/docs/permissionsandroid.html

**Comment by alvelig** [comment]
We'll have to implement this for android 27+

**Comment by TsarlinDomingo** [comment]
I'm getting exact error with the following:

```
ext {
    minSdkVersion = 16
    compileSdkVersion = 27
    targetSdkVersion = 27
    buildToolsVersion = "27.1.1"
    supportLibVersion = "27.1.1"
    googlePlayServicesVersion = "11.8.0"
    androidMapsUtilsVersion = "0.5+"
}
```

**Environment:**
```
"react": "^16.3.0-alpha.1",
"react-native": "^0.56.0",
"react-native-maps": "^0.22.1",
```

**AndroidManifest.xml**
<img width="716" alt="screenshot 2018-11-08 at 19 09 35" src="https://user-images.githubusercontent.com/36619494/48221421-f5e12d80-e389-11e8-8de2-91fe2f75a969.png">

**ERROR**
<img width="279" alt="screenshot 2018-11-08 at 19 11 18" src="https://user-images.githubusercontent.com/36619494/48221456-16a98300-e38a-11e8-9775-62fd7406b139.png">

**Comment by rborn** [comment]
@TsarlinDomingo you need to recompile the app after you change the AndroidManifest

**Comment by TsarlinDomingo** [comment]
> @TsarlinDomingo you need to recompile the app after you change the AndroidManifest

@rborn I did that and it still gives the same error.

**Comment by y4gamilight** [comment]
I think you made mistake same with me, I forgot set permission for emulator so it make crash when i try to run app. Check your device again.

**Comment by traviisd** [comment]
> @y4gamilight I think you made mistake in setting. In my case, I forgot set permission for emulator so it make crash when i try to run app. Check your device again.

I also ran into this issue, went to the app settings on the emulator and enabled permissions and this solved the issue. Also it might be worth noting to double check your [Google API key](https://cloud.google.com/docs/authentication/api-keys?hl=en&visit_id=636788056606635784-4229707766&rd=1) as well. I forgot to create a new restricted key for android and had my iOS restricted key set. Once I created the new key, all was well.

**Comment by rborn** [comment]
I’m closing this issue because it has been inactive for a while. This probably means that it is not reproducible or it has been fixed in a newer version. If it’s an enhancement and hasn’t been taken on for so long, then it seems no one has the time to implement this.

Please reopen if you still encounter this issue with the [latest stable version](https://github.com/react-native-community/react-native-maps/releases).

Thank you!

**Comment by ahmedfaraz587** [comment]
I got this error when i want to show user location in maps. I tried every thing but nothing work. After many tries i got solution then i decided to share with all developers. If it works please subscribe and share this video. https://www.youtube.com/watch?v=_mi_QNmXSwM
